### ISSUE TITLE
If multiple `source` lines are in the same file, all but the first one are ignored

### ISSUE THREAD

**Issue by ArrayBolt3** [koalaman/shellcheck#2989]
#### For bugs
- Rule Id (if any, e.g. SC1000): N/A
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0 (installed via Snap Store from edge channel)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- I **did not** try on https://www.shellcheck.net/ to verify that this is still a problem on the latest commit, as this bug cannot be tested on shellcheck.net. It requires multiple files. I may try to compile Shellcheck from source later on.

#### Here's a snippet or screenshot that shows the problem:

**Note all these files must be in the same dir.**

File: one.sh
```sh
#!/bin/bash
declare -g var1;
var1='abc';
```

File: two.sh
```sh
#!/bin/bash
declare -g var2;
var2='def';
```

File: script.sh
```sh
#!/bin/bash
# shellcheck source=one.sh
# shellcheck source=two.sh
. one.sh
. two.sh
echo "${var1}${var2}";
```

#### Here's what shellcheck currently says:

```
┌─╴user@kf-XE:~/test
└─╴$ shellcheck -x script.sh 

In script.sh line 6:
echo "${var1}${var2}";
             ^-----^ SC2154 (warning): var2 is referenced but not assigned (did you mean 'var1'?).

For more information:
  https://www.shellcheck.net/wiki/SC2154 -- var2 is referenced but not assign...
```

#### Here's what I wanted or expected to see:

No output.

#### Additional notes:

* `script.sh` runs fine and prints "abcdef" as expected.
* If I reverse the `shellcheck source` lines in `script.sh` so that `two.sh` is on top and `one.sh` is on the bottom, I get the reverse warning - now var1 is seen as referenced but not assigned rather than var2. The script still runs correctly and gives the expected output.

**Comment by nidorhuang** [comment]
I bumped into the same issue days before. However I figured out that the correct usage is to place the `# shellcheck source=sourced_file` right before each `source sourced_file` command, respectively. The example below works for me.

```
# shellcheck source=sourced_file1
source sourced_file1
# shellcheck source=sourced_file2
source sourced_file2
# shellcheck source=sourced_file3
source sourced_file3
```

**Comment by ArrayBolt3** [comment]
hmm, I'll try something like that. It'll have to work even if the source lines are put in places other than above the actual source command though because the scripts I'm working with use a "dynamic linking" system whereby a function is used to locate and source a script when given info about which script to source.

**Comment by ale5000-git** [comment]
@ArrayBolt3
Note that shellcheck will use the line `# shellcheck source=sourced_file1` by looking of the current folder where you are running the shellcheck command.

So if you have the files inside dir1 and run a command like this
```
shellcheck dir1/file1
```
it will fail instead with this will succeed:
```
cd dir1
shellcheck file1
```

To make it less error prone and just ignore the current folder you can put this in the file:
```
# shellcheck source=SCRIPTDIR/sourced_file1
source sourced_file1
```
to use the folder where the files is, instead of the current folder.
